lib/Page.js

```diff
@@ -644,7 +649,11 @@ class Page extends EventEmitter {
 
     const responses = new Map();
     const listener = helper.addEventListener(this._networkManager, NetworkManager.Events.Response, response => responses.set(response.url(), response));
-    const error = await watcher.navigationPromise();
+    const error = await Promise.race([
+      watcher.timeoutPromise(),
```

Comment:
why await the timeout promise here? Won't the same and new document navigations be called with the timeout?

Reply:
no, I split them out (see NavigationWatcher)